Formula/adios2.rb

```diff
@@ -0,0 +1,126 @@
+class Adios2 < Formula
+  desc "Next generation of ADIOS developed in the Exascale Computing Program"
+  homepage "https://adios2.readthedocs.io"
+  url "https://github.com/ornladios/ADIOS2/archive/v2.5.0.tar.gz"
+  sha256 "7c8ff3bf5441dd662806df9650c56a669359cb0185ea232ecb3578de7b065329"
+  head "https://github.com/ornladios/ADIOS2.git", :branch => "master"
+
+  depends_on "cmake" => :build
+  depends_on "gcc" => :build
+  depends_on "bzip2"
+  depends_on "c-blosc"
+  depends_on "catch2"
+  depends_on "hdf5"
+  depends_on "libfabric"
+  depends_on "libpng"
+  # mgard
+  depends_on "numpy"
+  # open-mpi
+  depends_on "python"
+  # sz
+  depends_on "zeromq"
+  # zfp
+
+  def install
+    args = std_cmake_args + %W[
+      -DADIOS2_USE_Blosc=ON
+      -DADIOS2_USE_BZip2=ON
+      -DADIOS2_USE_DataSpaces=OFF
+      -DADIOS2_USE_Fortran=ON
+      -DADIOS2_USE_HDF5=ON
+      -DADIOS2_USE_MGARD=OFF
+      -DADIOS2_USE_MPI=OFF
+      -DADIOS2_USE_PNG=ON
+      -DADIOS2_USE_Python=ON
+      -DADIOS2_USE_ZeroMQ=ON
+      -DADIOS2_USE_ZFP=OFF
+      -DCMAKE_DISABLE_FIND_PACKAGE_BISON=TRUE
+      -DCMAKE_DISABLE_FIND_PACKAGE_CrayDRC=TRUE
+      -DCMAKE_DISABLE_FIND_PACKAGE_FLEX=TRUE
+      -DCMAKE_DISABLE_FIND_PACKAGE_LibFFI=TRUE
+      -DCMAKE_DISABLE_FIND_PACKAGE_NVSTREAM=TRUE
+      -DPYTHON_EXECUTABLE:FILEPATH=#{Formula["python"].opt_bin}/python3
+      -DADIOS2_BUILD_TESTING=OFF
+      -DADIOS2_BUILD_EXAMPLES=OFF
+    ]
+    mkdir "build" do
+      system "cmake", "..", *args
+      system "make", "install"
+      rm_rf Dir[prefix/"bin/bp4dbg"] # https://github.com/ornladios/ADIOS2/pull/1846
+    end
+  end
+
+  test do
+    (testpath/"write.cpp").write <<~EOS
+      #include <adios2.h>
+      #include <iostream>
+      #include <ios>
+      #include <vector>
+      #include <stdexcept>
+
+      int main(int argc, char *argv[])
+      {
+        int rank = 0, nproc = 1;
+        const int NSTEPS = 5;
+        adios2::ADIOS adios;
+        const unsigned int Nx = 10;
+        std::vector<double> row(Nx);
+
+        adios2::IO io = adios.DeclareIO("Output");
+        adios2::Variable<double> varGlobalArray =
+          io.DefineVariable<double>("GlobalArray", {(unsigned int)nproc, Nx});
+        adios2::Engine writer = io.Open("globalArray.bp", adios2::Mode::Write);
+
+        for (size_t step = 0; step < NSTEPS; step++)
+        {
+          writer.BeginStep();
+
+          for (size_t i = 0; i < Nx; i++)
+          {
+            row[i] = step * Nx * nproc * 1.0 + rank * Nx * 1.0 + (double)i;
+          }
+          varGlobalArray.SetSelection(adios2::Box<adios2::Dims>(
+            {static_cast<size_t>(rank), 0}, {1, static_cast<size_t>(Nx)}));
+          writer.Put<double>(varGlobalArray, row.data());
+          writer.EndStep();
+        }
+        writer.Close();
+        return EXIT_SUCCESS;
+      }
+    EOS
+    system ENV.cxx, "-std=c++11", "write.cpp", "-I#{opt_include}", "-ladios2"
```

Comment:
The flags used here might be different, especially with MPI in the mix.  The recommended practice is to use the output of the installed `adios2-config --cxx`, which can be passed directly to the compiler and is how an end user with makefiles would use it.

Reply:
Note too that if you're using an mpi enabled adios2 then the flags from `adios2-config` assume you're using the mpi compiler wrappers so any flags added by `mpic++` are stripped from the output.

Reply:
Okay, sure! So probably something like:
```suggestion
    adios2_config_flags = `adios2-config --cxx`.chomp.split
    system ENV.cxx, "write.cpp", *adios2_config_flags
```